programs/bpf/rust/invoke/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+//! @brief Example Rust-based BPF program that issues a cross-program-invocation
+
+#![allow(unreachable_code)]
+
+extern crate solana_sdk;
+
+use solana_bpf_rust_invoked::instruction::create_instruction;
+use solana_sdk::{
+    account_info::AccountInfo, cross_program, entrypoint, entrypoint::ProgramResult, info,
+    program_error::ProgramError, pubkey::Pubkey,
+};
+
+// const MINT_INDEX: usize = 0;
+const ARGUMENT_INDEX: usize = 1;
+const INVOKED_PROGRAM_INDEX: usize = 2;
+const INVOKED_ARGUMENT_INDEX: usize = 3;
+const INVOKED_PROGRAM_DUP_INDEX: usize = 4;
+// const ARGUMENT_DUP_INDEX: usize = 5;
+const DERIVED_KEY_INDEX: usize = 6;
+const DERIVED_KEY2_INDEX: usize = 7;
+
+entrypoint!(process_instruction);
+fn process_instruction(
```

Comment:
Good idea will add those tests.  Also, there is currently no limit to the number of accounts that can be passed beside the memory restrictions of the calling process.  This could be a ddos vulnerability

Reply:
On second thought, is there a reason to do this?  The runtime only passes the accounts listed in the instruction to the invoked program.  Breaking up the accounts vector isn't a fun thing to do

Reply:
Ok maybe I'm missing something. What if a program is called with 2 account arguments and wants to call a program that needs 3?

Reply:
Then it fails, programs need to be passed all the accounts it needs to pass to any invokes calls it intends to make

Reply:
Sorry I didn't say that clearly. Couldnt the caller dupe one of its accounts to call the instruction that needs 3?

And the opposite case where you have an instruction that takes 3 accounts and wants to call an instruction that only accepts 2

Reply:
The runtime looks in the accounts slice for each account required by the instruction.  If the instruction needs an account that is not in the slice then fail.  Once all the accounts have been found they are run through the pre account verification to ensure that they are valid.

Reply:
Not confident I'm addressing the question you have

Reply:
Ah I think I figured out the disconnect!

I was imagining that the caller program could rearrange, filter, dupe accounts etc. and then pass that new account slice to the called function. But that's overkill because as you mentioned, the cross program instruction just needs to reference the indices of the accounts from the original account slice.

As it stands, the caller could still choose to pass only a subset of the accounts to the called program I think. I don't see a check to make sure that all accounts in pre_accounts are present in the accounts slice

Reply:
Is there a reason that the caller needs to pass the accounts slice back out through `sol_process_signed_instruction_`? Can't we check for them how accounts have been modified?

Reply:
Ohhh we are doing that. Sorry!!!

Reply:
Oh, no we aren't actually 😅 

Does this make sense? Do we need the caller to pass through all the account infos? Don't we already know where the accounts are memory mapped?

Reply:
This issue is summarized in #9711  and is planned as a followup change to the PR